### PR TITLE
[HF] #2.0.11: add option to skip auto adding chages to commit

### PR DESCRIPTION
**Pull request by vborshchov**
### What does this PR do?
Adds flag to skip auto adding all chages `git add -A` from working tree to staging area before commit

### CODE REVIEW

File: actions/save_action.rb [hunk 1]
```diff
@@ -14,8 +14,19 @@ def call(args, options)
 
       Remote::Br.update
 
-      if Code.has_changes?
-        cmds << "git add -A && git commit -a -m '#{commit_message(task_name, commit_message, options)}' #{Remote::Br.exists?(current_branch) ? " && git pull origin #{current_branch} " : nil} #{options[:no_push] ? nil : "&& git push origin #{current_branch}"}"
+      if options[:skip_staging]
```
Comment (ruslan-palahin) [inline actions/save_action.rb:17]: #1 move to a new method

File: hf [hunk 2]
```diff
@@ -54,6 +54,7 @@ def main
   options[:quiet] = flags.include?('--quiet') || flags.include?('--q')
   options[:yes] = flags.include?('--yes') || flags.include?('--y')
   options[:echo] = flags.include?('--echo')
+  options[:skip_staging] = flags.include?('--dry')
```
Comment (ruslan-palahin) [inline hf]: #2 think on option name. Examples: --na, --no-add, --skip-add, --sa, etc.

File: hf [hunk 3]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env ruby
 
-VERSION = '2.0.10'
+VERSION = '2.0.11'
```
Comment (ruslan-palahin) [inline hf:3]: #3 Let's try to add tests.

**Review by ruslan-palahin — CHANGES_REQUESTED** [review]
@vborshchov great work, few suggestions added

File: hf (hunk 3 above)
Comment (vborshchov) [reply hf:3]: Added two tests

File: hf (hunk 2 above)
Comment (vborshchov) [reply hf]: Probably --skip-add option name is more readable

File: actions/save_action.rb (hunk 1 above)
Comment (vborshchov) [reply actions/save_action.rb:17]: Done

File: context.rb [hunk 4]
```diff
@@ -33,12 +33,28 @@ def raw_list
 class Code
   class << self
     def has_changes?
-      status = `git status`
-      mapping = {
-          'ru' => 'нечего коммитить',
-          'en' => 'nothing to commit'
-      }
-      !status.include? mapping[Config.locale]
+      !`git status --porcelain`.to_s.strip.empty?
```
Comment (ruslan-palahin) [inline context.rb:36]: :heart_eyes: :

**Review by ruslan-palahin — APPROVED** [review]

**Pull request merged**